sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/implementation/batch/BulkExecutorTest.java

```diff
@@ -234,6 +239,337 @@ public void executeBulk_OnGoneFailure() throws InterruptedException {
+        String duplicatePK = UUID.randomUUID().toString();
+        String id = UUID.randomUUID().toString();
+        for (int i = 0; i < totalRequest; i++) {
+            if (i == 0) {
+                BatchTestBase.EventDoc eventDoc = new BatchTestBase.EventDoc(id, 2, 4, "type1",
+                    duplicatePK);
+                cosmosItemOperations.add(CosmosBulkOperations.getCreateItemOperation(eventDoc,
+                    new PartitionKey(duplicatePK)));
+            } else {
+                cosmosItemOperations.add(CosmosBulkOperations.getPatchItemOperation(id,
+                    new PartitionKey(duplicatePK),
+                    CosmosPatchOperations.create().replace("/type", "updated" + i)));
+            }
+        }
+
+        CosmosItemOperation[] itemOperationsArray =
+            new CosmosItemOperation[cosmosItemOperations.size()];
+        cosmosItemOperations.toArray(itemOperationsArray);
+        CosmosBulkExecutionOptions cosmosBulkExecutionOptions = new CosmosBulkExecutionOptions();
+        ImplementationBridgeHelpers.CosmosBulkExecutionOptionsHelper
+            .getCosmosBulkExecutionOptionsAccessor()
+            .setPreserveOrdering(cosmosBulkExecutionOptions, true);
+
+        Flux<CosmosItemOperation> inputFlux = Flux
+            .fromArray(itemOperationsArray)
+            .delayElements(Duration.ofMillis(100));
+        final BulkExecutorTemp<BulkExecutorTest> executor = new BulkExecutorTemp<>(
+            container,
+            inputFlux,
+            cosmosBulkExecutionOptions);
+
+
+        try {
+            List<CosmosBulkOperationResponse<BulkExecutorTest>> bulkResponse =
+                Flux.deferContextual(context -> executor.execute()).collect(Collectors.toList()).block();
+
+
+            assertThat(bulkResponse.size()).isEqualTo(totalRequest);
+
+            for (int i = 0; i < cosmosItemOperations.size(); i++) {
+                CosmosBulkOperationResponse<BulkExecutorTest> operationResponse = bulkResponse.get(i);
+                com.azure.cosmos.models.CosmosBulkItemResponse cosmosBulkItemResponse =
+                    operationResponse.getResponse();
+                if (i == 0) {
+                    assertThat(cosmosBulkItemResponse.getStatusCode()).isEqualTo(HttpResponseStatus.CREATED.code());
+                } else {
+                    assertThat(cosmosBulkItemResponse.getStatusCode()).isEqualTo(HttpResponseStatus.OK.code());
+                }
+                assertThat(operationResponse.getOperation()).isEqualTo(cosmosItemOperations.get(i));
+                assertThat(cosmosBulkItemResponse.getRequestCharge()).isGreaterThan(0);
+                assertThat(cosmosBulkItemResponse.getCosmosDiagnostics().toString()).isNotNull();
+                assertThat(cosmosBulkItemResponse.getSessionToken()).isNotNull();
+                assertThat(cosmosBulkItemResponse.getActivityId()).isNotNull();
+                assertThat(cosmosBulkItemResponse.getRequestCharge()).isNotNull();
```

Comment:
You are making a nullability check after doing a greater than 0 check. Nullability check should come earlier.

Reply:
changed